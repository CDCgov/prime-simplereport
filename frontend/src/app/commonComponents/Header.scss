@use "../../scss/settings" as *;

header.usa-header {
  background-color: $brand-darker;
}

.usa-nav-container.prime-header {
  max-width: $site-max-width;

  .prime-organization-name {
    color: $brand-lightest;
    font-size: $size-body-3xs;
  }

  .prime-nav-link {
    color: $brand-lightest;
    font-size: 18px;
    position: relative;
    display: inline-block;

    &:hover {
      color: $brand-lighter;
    }

    &.active-nav-item::after {
      content: "";
      position: absolute;
      left: 0;
      bottom: 2px;
      width: 100%;
      height: 4px;
      background-color: $brand-light;
    }
  }

  .prime-facility-select {
    @media (min-width: "desktop") {
      .usa-sidenav__item {
        padding: 0 1.1rem;
      }
    }
    margin: 0;

    select {
      margin: 0.3rem 0;
    }
  }

  // .usa-navbar {
  //   margin-right: 1rem;
  // }

  .role-tag {
    border-bottom: 1px #dfe1e2 solid;
    // $theme-color-prime-gray:
    color: #71767a;
    margin-top: 0.5rem;
    padding-bottom: 1rem;
  }

  .header-link-icon {
    // $theme-color-primary
    color: #005ea2;
    width: 16px;
    height: 16px;
    margin-right: 0.5rem;
  }

  .sparkle-icon-mask {
    background-color: #005ea2;
    mask: url("../../img/font-awesome-pro-icons/sparkles.svg") no-repeat center;
  }

  .ghost-user-link {
    display: block;
    padding: 0;
    margin-top: 0.75rem;
    // $theme-color-primary
    color: #005ea2 !important;
  }

  .ghost-user-link:hover {
    background-color: #fff;
  }

  @media (max-width: "desktop") {
    .usa-sidenav__item {
      border-top: none !important;
    }
  }

  .mobile-sublist-container {
    margin-top: 0;
    padding: 0.5rem 1rem;
  }

  .mobile-sublist-container .usa-sidenav__item {
    margin-top: 0.5rem;
    border-top: none;
  }

  .mobile-menu-action {
    margin-left: 0.5rem;
  }

  .nav__primary-item-icon:last-child {
    border-bottom: 1px solid #dfe1e2;
  }

  .mobile-secondary-nav-container {
    margin-top: 0;
    font-size: 1.2rem;
  }

  .mobile-secondary-nav-container a,
  .mobile-main-nav-container a {
    padding: 0.75rem 1rem;
  }

  .mobile-facility-select {
    margin-top: 0;
  }

  .mobile-facility-select select {
    width: 100%;
  }

  .navlink__support {
    display: flex;
    align-items: center;
    margin-top: 0.75rem;

    button {
      color: #005ea2;
      text-decoration: none;
      padding: 0 !important;
    }

    a {
      color: #005ea2 !important;
      font-weight: 500;
      padding: 0 !important;
    }

    a:hover,
    button:hover {
      color: #005ea2;
      text-decoration: underline;
      font-weight: 700;
      background-color: #fff;
    }

    a:visited {
      color: #005ea2;
    }
  }

  .facility-select-mobile-container {
    padding-bottom: 2rem;
  }
}
