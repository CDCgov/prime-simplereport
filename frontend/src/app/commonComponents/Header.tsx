import React, { useState } from "react";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import classNames from "classnames";
import { v4 as uuidv4 } from "uuid";
import { useSelector, connect } from "react-redux";
import {
  useAppInsightsContext,
  useTrackEvent,
} from "@microsoft/applicationinsights-react-js";

import { PATIENT_TERM_PLURAL_CAP } from "../../config/constants";
import { formatFullName } from "../utils/user";
import siteLogo from "../../img/simplereport-logo-color.svg";
import { hasPermission, appPermissions } from "../permissions";
import { RootState } from "../store";

import Button from "./Button/Button";
import Dropdown from "./Dropdown";
import useComponentVisible from "./ComponentVisible";
import { LinkWithQuery } from "./LinkWithQuery";
import { TrainingNotification } from "./TrainingNotification";
import ChangeUser from "./ChangeUser";

const Header: React.FC<{}> = () => {
  const appInsights = useAppInsightsContext();
  const trackSupport = useTrackEvent(appInsights, "Support", {});

  const handleSupportClick = (e: MouseEvent) => {
    if (appInsights) {
      trackSupport(e);
    }
  };

  const isAdmin = useSelector<RootState, boolean>(
    (state) => state.user.isAdmin
  );
  const organization = useSelector(
    (state) => (state as any).organization as Organization
  );
  const facilities = useSelector(
    (state) => ((state as any).facilities as Facility[]) || []
  );
  const facility = useSelector(
    (state) => ((state as any).facility as Facility) || { id: "", name: "" }
  );
  const user = useSelector((state) => (state as any).user as User);
  const [menuVisible, setMenuVisible] = useState(false);
  const {
    ref: staffDefailsRef,
    isComponentVisible: staffDetailsVisible,
    setIsComponentVisible: setStaffDetailsVisible,
  } = useComponentVisible(false);

  const onFacilitySelect = (e: React.FormEvent<HTMLSelectElement>) => {
    const id = (e.target as HTMLSelectElement).value;
    window.location.href = `${
      window.location.pathname
    }?facility=${encodeURIComponent(id)}`;
  };

  const canViewSettings = hasPermission(
    user.permissions,
    appPermissions.settings.canView
  );

  const canViewPeople = hasPermission(
    user.permissions,
    appPermissions.people.canView
  );

  const canViewResults = hasPermission(
    user.permissions,
    appPermissions.results.canView
  );

  const canViewTestQueue = hasPermission(
    user.permissions,
    appPermissions.tests.canView
  );

  const siteLogoLinkPath = isAdmin ? "/admin" : "/queue";

  const logout = () => {
    // Fetch the id_token from local storage
    const id_token = localStorage.getItem("id_token");
    const state = uuidv4();
    // Remove auth data from local_storage
    localStorage.removeItem("access_token");
    localStorage.removeItem("id_token");
    window.location.replace(
      "https://hhs-prime.okta.com/oauth2/default/v1/logout" +
        `?id_token_hint=${encodeURIComponent(id_token || "")}` +
        `&post_logout_redirect_uri=${encodeURIComponent(
          process.env.REACT_APP_BASE_URL || ""
        )}` +
        `&state=${state}`
    );
  };

  return (
    <header className="usa-header usa-header--basic">
      {process.env.REACT_APP_IS_TRAINING_SITE === "true" && (
        <TrainingNotification />
      )}
      <div className="usa-nav-container">
        <div className="usa-navbar">
          <div className="usa-logo" id="basic-logo">
            <LinkWithQuery to={siteLogoLinkPath} title="Home" aria-label="Home">
              <img
                className="width-card desktop:width-full"
                src={siteLogo}
                alt="{process.env.REACT_APP_TITLE}"
              />
            </LinkWithQuery>
            <div className="prime-organization-name">{organization.name}</div>
          </div>
          <button
            onClick={() => setMenuVisible(!menuVisible)}
            className="usa-menu-btn"
          >
            Menu
          </button>
        </div>

        <nav
          aria-label="Primary navigation"
          className={classNames(
            "usa-nav",
            "prime-nav",
            "desktop:display-none",
            {
              "is-visible": menuVisible,
            }
          )}
        >
          <button
            className="fa-layers fa-fw fa-2x usa-nav__close prime-nav-close-button"
            onClick={() => setMenuVisible(false)}
            title={"close menu"}
          >
            <FontAwesomeIcon icon={"window-close"} />
          </button>

          <ul className="usa-nav__primary usa-accordion">
            {canViewTestQueue ? (
              <li className="usa-nav__primary-item prime-staff-infobox-sidemenu prime-settings-hidden">
                <LinkWithQuery
                  to={`/queue`}
                  onClick={() => setMenuVisible(false)}
                  activeClassName="active-nav-item"
                  className="prime-nav-link"
                  activeStyle={{
                    color: "white",
                  }}
                >
                  Conduct Tests
                </LinkWithQuery>
              </li>
            ) : null}
            {canViewResults ? (
              <li className="usa-nav__primary-item prime-staff-infobox-sidemenu prime-settings-hidden">
                <LinkWithQuery
                  to={`/results`}
                  onClick={() => setMenuVisible(false)}
                  activeClassName="active-nav-item"
                  className="prime-nav-link"
                  activeStyle={{
                    color: "white",
                  }}
                >
                  Results
                </LinkWithQuery>
              </li>
            ) : null}
            {canViewPeople ? (
              <li className="usa-nav__primary-item prime-staff-infobox-sidemenu prime-settings-hidden">
                <LinkWithQuery
                  to={`/patients`}
                  onClick={() => setMenuVisible(false)}
                  activeClassName="active-nav-item"
                  className="prime-nav-link"
                  activeStyle={{
                    color: "white",
                  }}
                >
                  {PATIENT_TERM_PLURAL_CAP}
                </LinkWithQuery>
              </li>
            ) : null}
            <li className="usa-nav__primary-item prime-staff-infobox-sidemenu prime-settings-hidden">
              <FontAwesomeIcon
                icon={"user-circle"}
                style={{
                  fill: "white",
                }}
              />
            </li>

            <li className="usa-nav__primary-item usa-sidenav prime-staff-infobox-sidemenu prime-settings-hidden">
              <ul className="usa-sidenav__sublist prime-sidenav_inset">
                <li className="usa-sidenav__item span-full-name">
                  {formatFullName(user)}
                </li>
                <li className="usa-sidenav__item">
                  <span>
                    <strong>Role: </strong>
                    {user.roleDescription}
                  </span>
                </li>
                <li className="usa-sidenav__item">{facility.name}</li>
              </ul>
            </li>
            <div>
              <div className="navlink__support">
                <a
                  href="https://simplereport.gov/support"
                  target="none"
                  onClick={() => handleSupportClick}
                >
                  Support
                </a>
              </div>
              <Button variant="unstyled" label="Log out" onClick={logout} />
              <ChangeUser />
            </div>
            {canViewSettings ? (
              <li className="usa-nav__primary-item prime-settings-hidden">
                <LinkWithQuery
                  to={`/settings`}
                  onClick={() => setMenuVisible(false)}
                  activeClassName="active-nav-item"
                  activeStyle={{
                    color: "white",
                  }}
                >
                  <FontAwesomeIcon icon={"cog"} /> Settings
                </LinkWithQuery>
              </li>
            ) : null}
          </ul>
        </nav>

        <nav aria-label="Primary navigation" className="usa-nav prime-nav">
          <ul className="usa-nav__primary usa-accordion">
            {canViewTestQueue ? (
              <li className="usa-nav__primary-item">
                <LinkWithQuery
                  to={`/queue`}
                  onClick={() => setMenuVisible(false)}
                  activeClassName="active-nav-item"
                  className="prime-nav-link"
                  id="conduct-test-nav-link"
                  activeStyle={{
                    color: "white",
                  }}
                >
                  Conduct Tests
                </LinkWithQuery>
              </li>
            ) : null}
            {canViewResults ? (
              <li className="usa-nav__primary-item">
                <LinkWithQuery
                  to={`/results`}
                  onClick={() => setMenuVisible(false)}
                  activeClassName="active-nav-item"
                  className="prime-nav-link"
                  id="results-nav-link"
                  activeStyle={{
                    color: "white",
                  }}
                >
                  Results
                </LinkWithQuery>
              </li>
            ) : null}
            {canViewPeople ? (
              <li className="usa-nav__primary-item">
                <LinkWithQuery
                  to={`/patients`}
                  onClick={() => setMenuVisible(false)}
                  activeClassName="active-nav-item"
                  className="prime-nav-link"
                  id="patient-nav-link"
                  activeStyle={{
                    color: "white",
                  }}
                >
                  {PATIENT_TERM_PLURAL_CAP}
                </LinkWithQuery>
              </li>
            ) : null}
          </ul>
          {facilities && facilities.length > 0 ? (
            <div className="prime-facility-select">
              <Dropdown
                selectedValue={facility.id}
                onChange={onFacilitySelect}
                options={facilities.map(({ name, id }) => ({
                  label: name,
                  value: id,
                }))}
              />
            </div>
          ) : null}
          <ul className="usa-nav__primary usa-accordion">
            <li className="usa-nav__primary-item nav__primary-item-icon">
              <LinkWithQuery
                to={`#`}
                isActive={() => staffDetailsVisible}
                onClick={(e) => {
                  e.preventDefault();
                  setStaffDetailsVisible(!staffDetailsVisible);
                }}
                activeClassName="active-nav-item"
                data-testid="user-button"
              >
                <FontAwesomeIcon
                  icon={"user-circle"}
                  style={{
                    color: staffDetailsVisible ? "white" : "",
                  }}
                />
              </LinkWithQuery>
              <div
                ref={staffDefailsRef}
                aria-label="Primary navigation"
                className={classNames("shadow-3", "prime-staff-infobox", {
                  "is-prime-staff-infobox-visible": staffDetailsVisible,
                })}
              >
                <ul className="usa-sidenav__sublist">
                  <li className="usa-sidenav__item span-full-name">
                    {formatFullName(user)}
                  </li>
                  <li className="usa-sidenav__item">
                    <span>
                      <strong>Role: </strong>
                      {user.roleDescription}
                    </span>
                  </li>
                  <li className="usa-sidenav__item">{facility.name}</li>
                  <li className="usa-sidenav__item navlink__support">
                    <a
                      href="https://simplereport.gov/support"
                      target="none"
                      onClick={() => handleSupportClick}
                      data-testid="support-link"
                    >
                      Support
                    </a>
                  </li>
                  <li className="usa-sidenav__item margin-top-2">
                    <Button
                      variant="unstyled"
                      label=" Log out"
                      onClick={logout}
                    />
                  </li>
                  <ChangeUser />
                </ul>
              </div>
            </li>
            {canViewSettings ? (
              <li className="usa-nav__primary-item nav__primary-item-icon">
                <LinkWithQuery
                  to={`/settings`}
                  onClick={() => setMenuVisible(false)}
                  activeClassName="active-nav-item"
                  activeStyle={{
                    color: "white",
                  }}
                >
                  <FontAwesomeIcon icon={"cog"} />
                </LinkWithQuery>
              </li>
            ) : null}
          </ul>
        </nav>
      </div>
    </header>
  );
};

export default connect()(Header);
