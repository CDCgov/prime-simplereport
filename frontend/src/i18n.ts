import i18n from "i18next";
import { initReactI18next } from "react-i18next";

import { en } from "./lang/en";
import { es } from "./lang/es";
import { eo } from "./lang/eo";

export const resources = { en, es, eo };

i18n.use(initReactI18next).init({
  resources,
  lng: localStorage.getItem("language") || "en",
  interpolation: {
    escapeValue: false,
  },
});

export default i18n;
