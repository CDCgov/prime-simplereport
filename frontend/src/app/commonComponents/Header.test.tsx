import { render } from "@testing-library/react";
import "../../i18n";
import { Provider } from "react-redux";
import { MemoryRouter } from "react-router-dom";
import { MockedProvider } from "@apollo/client/testing";
import configureStore from "redux-mock-store";
import { configureAxe } from "jest-axe";

import Header from "./Header";

const axe = configureAxe({
  rules: {
    // disable landmark rules when testing isolated components.
    region: { enabled: false },
  },
});
const mockStore = configureStore([]);
const store = mockStore({
  organization: {
    name: "Organization Name",
  },
  user: {
    firstName: "Kim",
    lastName: "Mendoza",
    roleDescription: "Admin user",
  },
  facilities: [
    { id: "1", name: "Facility One" },
    { id: "2", name: "Facility Two" },
  ],
});

const mockNavigate = jest.fn();
jest.mock("react-router-dom", () => {
  const original = jest.requireActual("react-router-dom");
  return {
    ...original,
    useNavigate: () => mockNavigate,
  };
});

describe("Header", () => {
  it("displays correctly", async () => {
    render(
      <Provider store={store}>
        <MockedProvider>
          <MemoryRouter>
            <Header />
          </MemoryRouter>
        </MockedProvider>
      </Provider>
    );
    expect(document.body).toMatchSnapshot();
    expect(await axe(document.body)).toHaveNoViolations();
  });
});
