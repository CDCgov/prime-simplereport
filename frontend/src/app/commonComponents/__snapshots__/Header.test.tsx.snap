// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header displays correctly 1`] = `
<body>
  <div>
    <div
      class="usa-nav-container prime-header"
    >
      <div
        class="usa-navbar"
      >
        <div
          class="usa-logo"
          id="basic-logo"
        >
          <a
            aria-label="Home"
            class=""
            href="/queue"
            title="Home"
          >
            <img
              alt="SimpleReport"
              class="width-card desktop:width-full"
              src="simplereport-logo-white.svg"
            />
          </a>
          <div
            class="prime-organization-name margin-left-4"
          >
            Organization Name
          </div>
        </div>
        <button
          class="usa-menu-btn"
        >
          Menu
        </button>
        <nav
          aria-label="Primary mobile navigation"
          class="usa-nav prime-nav desktop:display-none mobile-nav"
        >
          <button
            class="fa-layers fa-fw fa-2x usa-nav__close prime-nav-close-button"
            title="close menu"
          >
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-rectangle-xmark "
              data-icon="rectangle-xmark"
              data-prefix="fas"
              focusable="false"
              role="img"
              viewBox="0 0 512 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M64 32C28.7 32 0 60.7 0 96V416c0 35.3 28.7 64 64 64H448c35.3 0 64-28.7 64-64V96c0-35.3-28.7-64-64-64H64zM175 175c9.4-9.4 24.6-9.4 33.9 0l47 47 47-47c9.4-9.4 24.6-9.4 33.9 0s9.4 24.6 0 33.9l-47 47 47 47c9.4 9.4 9.4 24.6 0 33.9s-24.6 9.4-33.9 0l-47-47-47 47c-9.4 9.4-24.6 9.4-33.9 0s-9.4-24.6 0-33.9l47-47-47-47c-9.4-9.4-9.4-24.6 0-33.9z"
                fill="currentColor"
              />
            </svg>
          </button>
          <ul
            class="usa-nav__primary usa-accordion mobile-secondary-nav-container"
          >
            <li
              class="usa-nav__primary-item nav__primary-item-icon"
            >
              <a
                class="prime-nav-link"
                data-testid="mobile-settings-button"
                href="/settings/users/1"
                id="mobile-settings-button"
                role="link"
              >
                Settings
              </a>
            </li>
          </ul>
          <div
            class="usa-nav__primary mobile-sublist-container"
          >
            <ul
              class="usa-sidenav__sublist"
            >
              <li
                class="usa-sidenav__item span-full-name"
              >
                Kim Mendoza
              </li>
              <li
                class="usa-sidenav__item role-tag"
              >
                Admin
              </li>
              <li
                class="usa-sidenav__item navlink__support"
              >
                <div
                  class="header-link-icon sparkle-icon-mask"
                />
                <a
                  data-testid="mobile-whats-new-link"
                  href="https://www.simplereport.gov/using-simplereport/whats-new"
                  rel="noreferrer"
                  target="_blank"
                >
                  What's new
                </a>
              </li>
              <li
                class="usa-sidenav__item navlink__support"
              >
                <svg
                  aria-hidden="true"
                  class="svg-inline--fa fa-circle-question header-link-icon"
                  data-icon="circle-question"
                  data-prefix="fas"
                  focusable="false"
                  role="img"
                  viewBox="0 0 512 512"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M256 512A256 256 0 1 0 256 0a256 256 0 1 0 0 512zM169.8 165.3c7.9-22.3 29.1-37.3 52.8-37.3h58.3c34.9 0 63.1 28.3 63.1 63.1c0 22.6-12.1 43.5-31.7 54.8L280 264.4c-.2 13-10.9 23.6-24 23.6c-13.3 0-24-10.7-24-24V250.5c0-8.6 4.6-16.5 12.1-20.8l44.3-25.4c4.7-2.7 7.6-7.7 7.6-13.1c0-8.4-6.8-15.1-15.1-15.1H222.6c-3.4 0-6.4 2.1-7.5 5.3l-.4 1.2c-4.4 12.5-18.2 19-30.6 14.6s-19-18.2-14.6-30.6l.4-1.2zM224 352a32 32 0 1 1 64 0 32 32 0 1 1 -64 0z"
                    fill="currentColor"
                  />
                </svg>
                <a
                  data-testid="mobile-support-link"
                  href="https://www.simplereport.gov/support"
                  target="none"
                >
                  Support
                </a>
              </li>
              <li
                class="usa-sidenav__item navlink__support"
              >
                <svg
                  aria-hidden="true"
                  class="svg-inline--fa fa-right-from-bracket header-link-icon"
                  data-icon="right-from-bracket"
                  data-prefix="fas"
                  focusable="false"
                  role="img"
                  viewBox="0 0 512 512"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M377.9 105.9L500.7 228.7c7.2 7.2 11.3 17.1 11.3 27.3s-4.1 20.1-11.3 27.3L377.9 406.1c-6.4 6.4-15 9.9-24 9.9c-18.7 0-33.9-15.2-33.9-33.9l0-62.1-128 0c-17.7 0-32-14.3-32-32l0-64c0-17.7 14.3-32 32-32l128 0 0-62.1c0-18.7 15.2-33.9 33.9-33.9c9 0 17.6 3.6 24 9.9zM160 96L96 96c-17.7 0-32 14.3-32 32l0 256c0 17.7 14.3 32 32 32l64 0c17.7 0 32 14.3 32 32s-14.3 32-32 32l-64 0c-53 0-96-43-96-96L0 128C0 75 43 32 96 32l64 0c17.7 0 32 14.3 32 32s-14.3 32-32 32z"
                    fill="currentColor"
                  />
                </svg>
                <button
                  class="usa-button usa-button--unstyled"
                  type="button"
                >
                   Log out
                </button>
              </li>
            </ul>
            <label
              class="usa-label "
              id="mobile-facility-label"
            >
              Facility
            </label>
            <div
              class="prime-facility-select facility-select-mobile-container"
            >
              <div
                class="usa-form-group prime-dropdown  mobile-facility-select"
              >
                <select
                  aria-label="Select testing facility"
                  aria-required="false"
                  class="usa-select"
                  id="1"
                >
                  <option
                    aria-selected="false"
                    value="1"
                  >
                    Facility One
                  </option>
                  <option
                    aria-selected="false"
                    value="2"
                  >
                    Facility Two
                  </option>
                </select>
              </div>
            </div>
            <button
              class="usa-button touchpoints-wrapper sr-app-touchpoints-button"
              type="button"
            >
              How can we improve SimpleReport?
            </button>
          </div>
        </nav>
      </div>
      <nav
        aria-label="Primary desktop navigation"
        class="usa-nav prime-nav desktop-nav"
      >
        <div
          class="prime-facility-select"
        >
          <div
            class="usa-form-group prime-dropdown "
          >
            <select
              aria-label="Select testing facility"
              aria-required="false"
              class="usa-select"
              id="2"
            >
              <option
                aria-selected="false"
                value="1"
              >
                Facility One
              </option>
              <option
                aria-selected="false"
                value="2"
              >
                Facility Two
              </option>
            </select>
          </div>
        </div>
        <ul
          class="usa-nav__primary usa-accordion"
        >
          <li
            class="usa-nav__primary-item nav__primary-item-icon"
          >
            <a
              aria-controls="desktop-user-button-submenu"
              aria-current="page"
              aria-expanded="false"
              class="prime-nav-link active"
              data-testid="desktop-user-button"
              href="/"
              id="desktop-user-button"
              role="button"
            >
              <svg
                aria-hidden="false"
                aria-label="My account"
                class="svg-inline--fa fa-circle-user "
                data-icon="circle-user"
                data-prefix="fas"
                focusable="false"
                role="img"
                viewBox="0 0 512 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M399 384.2C376.9 345.8 335.4 320 288 320H224c-47.4 0-88.9 25.8-111 64.2c35.2 39.2 86.2 63.8 143 63.8s107.8-24.7 143-63.8zM0 256a256 256 0 1 1 512 0A256 256 0 1 1 0 256zm256 16a72 72 0 1 0 0-144 72 72 0 1 0 0 144z"
                  fill="currentColor"
                />
              </svg>
            </a>
          </li>
          <li
            class="usa-nav__primary-item nav__primary-item-icon"
          >
            <a
              class="prime-nav-link"
              data-testid="desktop-settings-button"
              href="/settings/users/1"
              id="desktop-settings-button"
              role="link"
            >
              <svg
                aria-hidden="false"
                aria-label="Settings"
                class="svg-inline--fa fa-gear "
                data-icon="gear"
                data-prefix="fas"
                focusable="false"
                role="img"
                viewBox="0 0 512 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M495.9 166.6c3.2 8.7 .5 18.4-6.4 24.6l-43.3 39.4c1.1 8.3 1.7 16.8 1.7 25.4s-.6 17.1-1.7 25.4l43.3 39.4c6.9 6.2 9.6 15.9 6.4 24.6c-4.4 11.9-9.7 23.3-15.8 34.3l-4.7 8.1c-6.6 11-14 21.4-22.1 31.2c-5.9 7.2-15.7 9.6-24.5 6.8l-55.7-17.7c-13.4 10.3-28.2 18.9-44 25.4l-12.5 57.1c-2 9.1-9 16.3-18.2 17.8c-13.8 2.3-28 3.5-42.5 3.5s-28.7-1.2-42.5-3.5c-9.2-1.5-16.2-8.7-18.2-17.8l-12.5-57.1c-15.8-6.5-30.6-15.1-44-25.4L83.1 425.9c-8.8 2.8-18.6 .3-24.5-6.8c-8.1-9.8-15.5-20.2-22.1-31.2l-4.7-8.1c-6.1-11-11.4-22.4-15.8-34.3c-3.2-8.7-.5-18.4 6.4-24.6l43.3-39.4C64.6 273.1 64 264.6 64 256s.6-17.1 1.7-25.4L22.4 191.2c-6.9-6.2-9.6-15.9-6.4-24.6c4.4-11.9 9.7-23.3 15.8-34.3l4.7-8.1c6.6-11 14-21.4 22.1-31.2c5.9-7.2 15.7-9.6 24.5-6.8l55.7 17.7c13.4-10.3 28.2-18.9 44-25.4l12.5-57.1c2-9.1 9-16.3 18.2-17.8C227.3 1.2 241.5 0 256 0s28.7 1.2 42.5 3.5c9.2 1.5 16.2 8.7 18.2 17.8l12.5 57.1c15.8 6.5 30.6 15.1 44 25.4l55.7-17.7c8.8-2.8 18.6-.3 24.5 6.8c8.1 9.8 15.5 20.2 22.1 31.2l4.7 8.1c6.1 11 11.4 22.4 15.8 34.3zM256 336a80 80 0 1 0 0-160 80 80 0 1 0 0 160z"
                  fill="currentColor"
                />
              </svg>
            </a>
          </li>
        </ul>
      </nav>
    </div>
  </div>
</body>
`;
